CMake/json-download.cmake.in

```diff
@@ -5,15 +5,16 @@ include(ExternalProject)
 ExternalProject_Add(
     nlohmann_json
     PREFIX .
-    GIT_REPOSITORY "https://github.com/nlohmann/json.git"
-    GIT_TAG        v3.11.3
-    GIT_CONFIG     advice.detachedHead=false  # otherwise we'll get "You are in 'detached HEAD' state..."
-    SOURCE_DIR     "${CMAKE_BINARY_DIR}/third-party/json"
-    GIT_SHALLOW    1        # No history needed (requires cmake 3.6)
+    
+    # We do not use 'GIT_REPOSITORY' as it doesn't support a shallow clone of a specific commit,
```

Comment:
Please point to the article you pointed to in the PR

Reply:
And also explain `advice.detachedHead=false`

Reply:
Done